isis/tests/DistanceTests.cpp

```diff
@@ -0,0 +1,295 @@
+	EXPECT_EQ(dist.pixels(2), 1500500);
+	EXPECT_EQ(dist.meters(), 750250);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 750.25);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.0010779609);
+}
+
+
+TEST(DistanceTests, CopyConstructor) {
+	Distance origDist(1500.5, Distance::Meters);
+	Distance copiedDist(origDist);
+	ASSERT_FLOAT_EQ(copiedDist.meters(), 1500.5);
+}
+
+
+TEST(DistanceTests, SetMeters) {
+	Distance dist;
+	dist.setMeters(1500500);
+	ASSERT_EQ(dist.meters(), 1500500);
+}
+
+
+TEST(DistanceTests, SetKilometers) {
+	Distance dist;
+	dist.setKilometers(1500500);
+	ASSERT_EQ(dist.kilometers(), 1500500);
+}
+
+
+TEST(DistanceTests, SetSolarRadii) {
+	Distance dist;
+	dist.setSolarRadii(2);
+	ASSERT_EQ(dist.solarRadii(), 2);
+}
+
+
+TEST(DistanceTests, SetPixels) {
+	Distance dist;
+	dist.setPixels(1500500, 2);
+	ASSERT_EQ(dist.pixels(2), 1500500);
+}
+
+
+TEST(DistanceTests, SetNegativeDistance) {
+	try {
+		Distance dist;
+		dist.setMeters(-1);
+		FAIL() << "Expected error message: Negative distances are not supported";
+	}
+	catch(IException &e) {
+	    EXPECT_TRUE(e.toString().contains("Negative distances are not supported")) 
+	    									<< e.toString().toStdString();
+	}
+	catch(...) {
+	   	FAIL() << "Expected error message: Negative distances are not supported";
+	}
+}
+
+
+// TEST(DistanceTests, ToString) {
+// 	Distance dist(1500500, Distance::Meters);
+// 	ASSERT_EQ(dist.toString(), "1500500 meters");
+// }
+
+
+TEST(DistanceTests, IsValidTrue) {
+	Distance dist(1500500, Distance::Meters);
+	ASSERT_TRUE(dist.isValid());
+}
+
+
+TEST(DistanceTests, IsValidFalse) {
+	Distance dist;
+	ASSERT_FALSE(dist.isValid());
+}
+
+
+//should we do all units?
+TEST(DistanceTests, GreaterThanDifferent) {
+	Distance dist1(10, Distance::Meters);
+	Distance dist2(20, Distance::Meters);
+	EXPECT_TRUE(dist2 > dist1);
+	EXPECT_FALSE(dist1 > dist2);
+}
+
+
+TEST(DistanceTests, GreaterThanEqual) {
+	Distance dist1(10, Distance::Meters);
+	Distance dist2(10, Distance::Meters);
+	EXPECT_FALSE(dist2 > dist1);
+	EXPECT_FALSE(dist1 > dist2);
+}
+
+
+TEST(DistanceTests, GreaterThanNull) {
+	try {
+		bool value = Distance() < Distance();
+		ASSERT_TRUE(value);
+		FAIL() << "Expected error message: Distance has not been initialized";
+	}
+	catch(IException &e) {
+	    EXPECT_TRUE(e.toString().contains("Distance has not been initialized")) 
+	    									<< e.toString().toStdString();
+	}
+	catch(...) {
+	   	FAIL() << "Expected error message: Distance has not been initialized";
+	}
+}
+
+
+TEST(DistanceTests, LessThanDifferent) {
+	Distance dist1(10, Distance::Meters);
+	Distance dist2(20, Distance::Meters);
+	EXPECT_TRUE(dist1 < dist2);
+	EXPECT_FALSE(dist2 < dist1);
+}
+
+
+TEST(DistanceTests, LessThanEqual) {
+	Distance dist1(10, Distance::Meters);
+	Distance dist2(10, Distance::Meters);
+	EXPECT_FALSE(dist1 < dist2);
+	EXPECT_FALSE(dist2 < dist1);
+}
+
+
+TEST(DistanceTests, LessThanNull) {
+	try {
+		bool value = Distance() < Distance();
+		ASSERT_TRUE(value);
```

Comment:
What is this ASSERT_TRUE for if the comparison in the above line is supposed to throw an exception?

Reply:
I agree, this should be removed. We shouldn't be testing what that expression outputs because it throws an exception. If we are throwing an exception we should not also be honoring a specific return value.

Reply:
I forgot to make a comment about that when writing the test. The only reason why I instantiate a bool and use ASSERT_TRUE is so we do not get a build warning saying that it is unused. I would not have put it in otherwise. Would it be better to take it out and have the build warning?

Reply:
You can just remove the assignment and throw away the result of the comparison.

```
try {
  Distance() < Distance()
  FAIL() << "Failure Message"
}
```

Reply:
That is what I did at first and I get an "unused comparison" warning. We could turn the warnings off for this type.